var/spack/repos/builtin/packages/dftbplus/package.py

```diff
@@ -0,0 +1,140 @@
+
+        And the last thing we do here is to set the installdir
+        """
+        dircwd = os.getcwd()
+        makefile = FileFilter("makefile")
+        makefile.filter("ROOT := .*", "ROOT := {}".format(dircwd))
+
+        archmake = join_path(".", "sys", "make.x86_64-linux-gnu")
+        copy(archmake, join_path(dircwd, "make.arch"))
+
+        march = FileFilter(join_path(dircwd, 'make.arch'))
+
+        mconfig = FileFilter(join_path(dircwd, 'make.config'))
+
+        mconfig.filter('INSTALLDIR := .*', 'INSTALLDIR := {}'.format(prefix))
+
+        if '+gpu' in self.spec:
+            march.filter('MAGMADIR = .*',
+                         'MAGMADIR = {}'.format(spec['magma'].prefix))
+
+            mconfig.filter('WITH_GPU := .*', 'WITH_GPU := 1')
+
+        if '+mpi' in self.spec:
+            march.filter('SCALAPACKDIR = .*',
+                         'SCALAPACKDIR = {}'.format(spec['scalapack'].prefix))
+
+            march.filter('LIB_LAPACK = -l.*',
+                         'LIB_LAPACK = {}'.format(spec['blas'].libs.ld_flags))
+
+            march.filter('mpifort', '{}'.format(spec['mpi'].mpifc))
+
+            mconfig.filter('WITH_MPI := .*', 'WITH_MPI := 1')
+
+            if '+elsi' in self.spec:
+                mconfig.filter('WITH_ELSI := .*', 'WITH_ELSI := 1')
+
+                has_pexsi = 'enable_pexsi' in spec['elsi']
+
+                mconfig.filter('WITH_PEXSI := .*', 'WITH_PEXSI := {}'.format(
+                    '1' if has_pexsi is True else '0'
+                ))
+
+                march.filter("ELSIINCDIR .*", "ELSIINCDIR = {}".format(
+                    spec['elsi'].prefix.include
+                ))
+
+                march.filter("ELSIDIR .*",
+                             "ELSIDIR = {}".format(spec['elsi'].prefix))
+
+        else:
+            march.filter('LIB_LAPACK += -l.*',
+                         'LIB_LAPACK += {}'.format(spec['blas'].libs.ld_flags))
+
+        if '+sockets' in self.spec:
+            mconfig.filter('WITH_SOCKETS := .*', 'WITH_SOCKETS := 1')
+
+        if '+transport' in self.spec:
+            mconfig.filter('WITH_TRANSPORT := .*', 'WITH_TRANSPORT := 1')
+
+        if '+arpack' in self.spec:
+            march.filter('ARPACK_LIBS = .*', 'ARPACK_LIBS = {}'.format(
+                spec['arpack-ng'].libs.ld_flags
+            ))
+
+            mconfig.filter('WITH_ARPACK := .*', 'WITH_ARPACK := 1')
+
+        if '+dftd3' in self.spec:
+            march.filter('COMPILE_DFTD3 = .*', 'COMPILE_DFTD3 = 0')
+            march.filter('DFTD3_INCS = .*', 'DFTD3_INCS = -I{}'.format(
+                spec['dftd3-lib'].prefix.include
+            ))
+
+            march.filter('DFTD3_LIBS = .*', 'DFTD3_LIBS = -L{} -ldftd3'.format(
+                spec['dftd3-lib'].prefix
+            ))
+
+            mconfig.filter('WITH_DFTD3 := .*', 'WITH_DFTD3 := 1')
+
+        # Download some data needed for the tests
+        Executable('./utils/get_opt_externals')
```

Comment:
This doesn't actually run the script, you need:
```suggestion
        Executable('./utils/get_opt_externals')()
```